src/main/java/com/esri/samples/scene/sync_map_and_scene_viewpoints/SyncMapAndSceneViewpoints.java

```diff
@@ -0,0 +1,138 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package com.esri.samples.scene.sync_map_and_scene_viewpoints;
+
+import java.util.ArrayList;
+
+import javafx.application.Application;
+import javafx.geometry.Orientation;
+import javafx.scene.Scene;
+import javafx.scene.control.SplitPane;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.ArcGISScene;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.GeoView;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.mapping.view.SceneView;
+
+public class SyncMapAndSceneViewpoints extends Application {
+
+  private MapView mapView;
+  private SceneView sceneView;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+
+    // create split pane and JavaFX app scene
+    SplitPane splitPane = new SplitPane();
+    splitPane.setOrientation(Orientation.HORIZONTAL);
+    Scene fxScene = new Scene(splitPane);
+
+    // set title, size, and add JavaFX scene to stage
+    stage.setTitle("Sync Map and Scene Viewpoints");
+    stage.setWidth(1000);
+    stage.setHeight(700);
+    stage.setScene(fxScene);
+    stage.show();
+
+    // create a map and add a basemap to it
+    ArcGISMap map = new ArcGISMap();
+    map.setBasemap(Basemap.createImagery());
+
+    // create a scene and add a basemap to it
+    ArcGISScene scene = new ArcGISScene();
+    scene.setBasemap(Basemap.createImagery());
+
+    // set the map to a map view
+    mapView = new MapView();
+    mapView.setMap(map);
+
+    // set the scene to a scene view
+    sceneView = new SceneView();
+    sceneView.setArcGISScene(scene);
+
+    // add the map view and scene view to the split plane
+    splitPane.getItems().addAll(mapView, sceneView);
+
+    // add a view point changed listener to the map view
+    mapView.addViewpointChangedListener(viewpointChangedEvent -> {
+      synchronizeViewpoints(mapView);
+    });
+
+    // add a view point changed listener to the scene view
+    sceneView.addViewpointChangedListener(viewpointChangedEvent -> {
+      synchronizeViewpoints(sceneView);
+    });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Synchronizes the viewpoint across Geoviews.
+   */
+  private void synchronizeViewpoints(GeoView geoView) {
+
+    // if the user is actively navigating the GeoView, get its current viewpoint and set the other view to the same viewpoint
+    if (geoView.isNavigating()) {
+
+      Viewpoint geoViewPoint = geoView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE);
+
+      // create a list of geoviews being navigated
+      ArrayList<GeoView> arrayListofGeoView = new ArrayList<>();
```

Comment:
This list will be created every time the viewpoint changes even though the list never changes. You could create it in the start method and reference it here.